eth/typing.py

```diff
@@ -0,0 +1,26 @@
+from typing import (
+    Dict,
+    Iterable,
+    Tuple,
+    Union,
+)
+
+from eth_typing import (
+    Address,
+)
+from mypy_extensions import (
+    TypedDict,
+)
+
+
+# TODO: Move into eth_typing
+
+AccountDetails = TypedDict('AccountDetails',
```

Comment:
Ooooh!  `TypedDict` is cool.

Reply:
Yep, they let us avoid sprinkling the code with `assert` statements https://github.com/ethereum/py-evm/pull/1404#discussion_r226073247